a_sent_turtle_2.py

```diff
@@ -0,0 +1,90 @@
+class Message:
+    """
+        A Message class. Allows users to save the message details.
+    """
+    def __init__(self, message_id: int, message_title: str, message_body: str, sender: str):
+        self.id = message_id
+        self.title = message_title
+        self.body = message_body
+        self.sender = sender
+        self.was_read = False
+
+    def __str__(self) -> str:
+        """
+            The function returns a string containing the message and its details.
+            :return: A string containing the message and its details.
+        """
+        return 'Message #' + str(self.id) + ', from ' + self.sender + '. ' +\
+               self.title + ': ' + self.body
+
+    def len(self) -> int:
```

Comment:
similar to str len is also a magic method.
https://www.analyticsvidhya.com/blog/2021/08/explore-the-magic-methods-in-python/#:~:text=__len__%20magic%20method%20is%20used%20to%20find%20the,attribute%20which%20is%20usually%20containers.&text=When%20we%20invoke%20the%20len,the%20__len__%20method.